### PR TITLE
Bug fix/parallel creation of indexes incluster

### PR DESCRIPTION
**Pull request by kvahed**

### CODE REVIEW

File: arangod/Cluster/ClusterInfo.cpp [hunk 1]
```diff
@@ -1712,10 +1713,34 @@ int ClusterInfo::ensureIndexCoordinator(
   }
   std::string const idString = arangodb::basics::StringUtils::itoa(iid);
 
+  AgencyComm ac;
   int errorCode;
   try {
-    errorCode = ensureIndexCoordinatorWithoutRollback(
-      databaseName, collectionID, idString, slice, create, compare, resultBuilder, errorMsg, timeout);
+    auto start = std::chrono::steady_clock::now();
+    while (true) { // Keep trying for 2 minutes, if it's preconditions, which are
```
Comment (jsteemann) [inline arangod/Cluster/ClusterInfo.cpp]: I think the code is correct, but I don't understand this code comment. Can you fix it?

File: CHANGELOG [hunk 2]
```diff
@@ -7,6 +7,7 @@ v3.3.14 (XXXX-XX-XX)
 
 * fixed issue #6031: Broken LIMIT in nested list iterations 
 
+* fixed internal issue #2812: Cluster failes to create many indexes in parallel
```
Comment (jsteemann) [inline CHANGELOG]: IMHO it should be "fails" instead of "failes".

**Comment by kvahed** [comment]
Fixed typos incomplete comment and typo

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/536/

**Pull request merged**